BuildGenerated.sh

```diff
@@ -1,192 +1,50 @@
-                grep -v "diff --git" | \
-                grep -v -E '^index' | \
-                grep -v -E '[-+] "revision":' | \
-                grep -v -E '[-+] "etag":' ) ]]
-      then
-        echo "$discovery has only changed revision; reverting"
-        git checkout -q -- $discovery
-      fi   
-    done
-  fi
-fi
-
-if [ -z ${SKIPGENERATE+x} ];
+if [[ $1 == "" ]]
 then
-  if [[ $GENERATE_CHANGES_ONLY == "TRUE" ]]
-  then
-    # Only generate libraries for discovery docs that have changed or are new.
-    modified=$(git status -s -- $DISCOVERY_DOC_DIR | grep -E '\.json$' | grep -E '^ M' | cut "-d " -f3)
-    added=$(git status -s -- $DISCOVERY_DOC_DIR | grep -E '\.json$' | grep -E '^\?\?' | cut "-d " -f2)
-    needs_generation=(${modified[@]} ${added[@]})
-    # If we generate only a subset of the existing discoveries, we do so in a temporary
-    # folder so as not to delete generated code for the unmodified discoveries.
-    CODE_GENERATION_DIR=$TEMP_CODE_GENERATION_DIR
-  else
-    needs_generation=$(find $DISCOVERY_DOC_DIR -name '*.json')
-    CODE_GENERATION_DIR=$PERM_CODE_GENERATION_DIR
-  fi
-  # Delete all generated code
-  echo Deleting existing \'$CODE_GENERATION_DIR\' directory...
-  rm -rf $CODE_GENERATION_DIR
-  for jsonfile in ${needs_generation[@]}
-  do
-    IFS='/'; names=($jsonfile); unset IFS
-    name=$(echo ${names[-1]} | sed 's/.json//g')
-    case $name in
-      identitytoolkit_v3|oauth2_v1)
-        echo Ignoring: \'$name\'
-        ;;
-      *)
-        echo Generating: \'$name\'
-        run_csharp_generator "$jsonfile" "$CODE_GENERATION_DIR" features.json "EnumStorage/$name.json"
-        if [[ -f $(pwd)/PostGeneration/$name.sh ]]
-        then
-          echo "Running post-generation step for $name"
-          $(pwd)/PostGeneration/$name.sh "$CODE_GENERATION_DIR"
-        fi
-        ;;
-    esac
-  done
+  echo "Usage options:"
+  echo "  BuildGenerated.sh <file1> [file2]"
+  echo "  BuildGenerated.sh @file-with-paths"
+  echo "Each file (either in file-with-paths or specified directly)"
+  echo "should be a directory containing a project file, or a Discovery doc"
+  exit 1
 fi
 
-if [ -z ${SKIPBUILD+x} ]; then
-  rm -f Generated.sln
-  dotnet new sln --name Generated
-  echo $CODE_GENERATION_DIR/*/*.csproj | xargs dotnet sln Generated.sln add
-  dotnet restore Generated.sln
-  dotnet build Generated.sln --configuration $BUILD_CONFIGURATION --no-restore
-fi
+echo "Creating solution"
 
-if [ -z ${SKIPPACK+x} ]; then
-  # Delete all generated nupkgs
-  echo Deleting existing \'$NUPKG_DIR\' directory...
-  rm -rf $NUPKG_DIR
-  dotnet pack Generated.sln --configuration $BUILD_CONFIGURATION --no-restore --no-build --output $NUPKG_DIR
-fi
+# First argument is the generation directory, used for when Discovery docs have been specified.
+PROJECTS=$(dotnet run --project Src/Tools/BuildGeneratedArgumentTranslator -- Src/Generated $*)
+rm -rf Generated.sln
+dotnet new sln --name Generated
```

Comment:
The new solution will only contain the newly generated packages, right? I think this was not the case before.

Reply:
I think it was the case before, because we add everything from "wherever we were generating to", which was TempGenerated usually.